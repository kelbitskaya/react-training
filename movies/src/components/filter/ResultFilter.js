import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Constants from '../constants';

export default function ResultFilter(props) {
  const genres = Constants.GENRE;
  const { updateMovies } = props;
  const [activeGenre, setActiveGenre] = useState(genres[0]);

  const sortMovieList = (movieGenre) => {
    const movieList = Constants.MOVIES;
    if (movieGenre.id === genres[0].id) {
      updateMovies(movieList);
      setActiveGenre(genres[0]);
    } else {
      const currentGenre = movieGenre.title;

      const sortedMovieList = movieList.filter((movie) => movie.genre.some(
        (genre) => genre.toLowerCase === activeGenre,
      ));

      const activeGenreIndex = genres.findIndex((el) => el.title === currentGenre);
      setActiveGenre(genres[activeGenreIndex]);

      updateMovies(sortedMovieList);
    }
  };

  const filterMovies = (genre) => {
    sortMovieList(genre);
  };

  return (
    <ul className="genre-filter">
      { Constants.GENRE.map((genre) => (
        <li key={genre.id}>
          <button
            type="button"
            className={genre.title === activeGenre.title ? 'genre-filter__item active' : 'genre-filter__item'}
            key={genre.id}
            onClick={() => filterMovies(genre)}
          >
            {genre.title}
          </button>
        </li>
      ))}
    </ul>
  );
}

ResultFilter.propTypes = {
  updateMovies: PropTypes.func.isRequired,
};
