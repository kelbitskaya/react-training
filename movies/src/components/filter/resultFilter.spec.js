import React from 'react';
import ResultFilter from './ResultFilter';
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import rerender from "react-test-renderer";
import routeData from "react-router";

const mockLocation = {
  pathname: '/welcome',
  hash: '',
  search: '',
  state: ''
};

describe("ResultFilter component", () => {
  beforeEach(()=>{
    jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation)
  });

  test("should render ResultFilter component", ()=> {
    const initialStore = {};
    const store = configureStore([])(initialStore);
    const component = rerender.create(
      <Provider store={store}>
        <ResultFilter/>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ResultFilter has genre-filter ', () => {
    const initialStore = {};
    const store = configureStore([])(initialStore);
    const component = mount(
      <Provider store={store}>
        <ResultFilter/>
      </Provider>);

    const button = component.find(".genre-filter");
    expect(button.length).toBe(1);

  });
});
