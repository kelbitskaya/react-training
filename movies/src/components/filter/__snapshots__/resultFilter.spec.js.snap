// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultFilter component should render ResultFilter component 1`] = `
<ul
  className="genre-filter"
>
  <li>
    <button
      className="genre-filter__item active"
      data-name="all"
      onClick={[Function]}
      type="button"
    >
      all
    </button>
  </li>
  <li>
    <button
      className="genre-filter__item"
      data-name="Documentary"
      onClick={[Function]}
      type="button"
    >
      Documentary
    </button>
  </li>
  <li>
    <button
      className="genre-filter__item"
      data-name="Comedy"
      onClick={[Function]}
      type="button"
    >
      Comedy
    </button>
  </li>
  <li>
    <button
      className="genre-filter__item"
      data-name="Horror"
      onClick={[Function]}
      type="button"
    >
      Horror
    </button>
  </li>
  <li>
    <button
      className="genre-filter__item"
      data-name="Crime"
      onClick={[Function]}
      type="button"
    >
      Crime
    </button>
  </li>
</ul>
`;

exports[`ResultFilter component should render ResultFilter component 2`] = `
<ul
  className="genre-filter"
>
  <li>
    <button
      className="genre-filter__item active"
      data-name="all"
      onClick={[Function]}
      type="button"
    >
      all
    </button>
  </li>
  <li>
    <button
      className="genre-filter__item"
      data-name="Documentary"
      onClick={[Function]}
      type="button"
    >
      Documentary
    </button>
  </li>
  <li>
    <button
      className="genre-filter__item"
      data-name="Comedy"
      onClick={[Function]}
      type="button"
    >
      Comedy
    </button>
  </li>
  <li>
    <button
      className="genre-filter__item"
      data-name="Horror"
      onClick={[Function]}
      type="button"
    >
      Horror
    </button>
  </li>
  <li>
    <button
      className="genre-filter__item"
      data-name="Crime"
      onClick={[Function]}
      type="button"
    >
      Crime
    </button>
  </li>
</ul>
`;

exports[`SortBy component should render SortBy component 1`] = `
<div
  className="sort-by"
>
  <p
    className="sort-by__text"
  >
    sort by
  </p>
  <div
    className="dropdown"
  >
    <button
      className="dropbtn "
      onClick={[Function]}
      type="button"
    >
      release date
    </button>
    <ul
      className="dropdown-content "
    >
      <li
        className="dropdown-content__item"
        datatype="release_date"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        type="button"
      >
        release date
      </li>
      <li
        className="dropdown-content__item"
        datatype="title"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        type="button"
      >
        title
      </li>
      <li
        className="dropdown-content__item"
        datatype="rating"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="presentation"
        type="button"
      >
        rating
      </li>
    </ul>
  </div>
</div>
`;
